### DISCUSSION TITLE
Feature discussion: Shuffle list of updates before selection

### DISCUSSION THREAD

**Discussion by jjtheo** [scala-steward-org/scala-steward#2736] [Ideas]
When using scala-steward with scala-steward-action on a large repository with lots of dependencies, 
limiting the number of updates at each run can create and endless loop of updating the same first libs. 

For example, there's 2 libs: `com.a` & `com.b`. `com.a` published a new version every two days.
If I run scala-steward every week, `com.a` will be the only one selected on each run. `com.b` will never be updated.

We could add an option in scala-steward to shuffle the list when selecting the next update to apply. This way, we would not loop on the same first updates of the list.

What do you think?

**Comment by alejandrohdezma** [comment]
Hey @jjtheo, sorry for the late response. For this case you could use the [`dependencyOverrides` feature from the `.scalasteward.conf` file](https://github.com/scala-steward-org/scala-steward/blob/main/docs/repo-specific-configuration.md). With this, you can change the frequency for some of your artifacts, so instead of being updated every day, they're updated at your own pace:

```conf
dependencyOverrides = [
  {
    dependency = { groupId = "com.b" },
    pullRequests = { frequency = "1 day" },
  },
  {
    dependency = { groupId = "com.a" },
    pullRequests = { frequency = "30 day" },
  }
]
```

**Reply by jjtheo** [reply]
Hey thank you for the suggestion, this should be enough !
Along with the `pullRequests.grouping` option, I'll be able to tweak the config to be less annoyed by the frequently updated libs
